### PR TITLE
Updated locksmithing script for trainer boxes

### PR DESCRIPTION
**Pull request by Hiinky**
Updated options across the board for locksmith training boxes to be supported by crossing-training without interrupting anyone's current settings.  Can be used independently or in t2 also.

add support for the daily charge boxes and incidental trainers / keepsake boxes from burgle.  I opted to keep live boxes handled by pick.lic specifically because it's kind of a beast and needs updating but folks really want to use the trainer boxes now since pet boxes are dead.  Settings in new locksmithing.lic will send args to pick.lic as needed.

Handles in order:
- Live boxes based on existing ;pick settings - selected first to clear inventory weight
- Permanent trainer boxes - unlimited or daily charges - selected second to use up available charges before using up consumables.
- Consumable trainer boxes such as burgled keepsake boxes, incidental items

### CODE REVIEW

File: locksmithing.lic [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+
+custom_require.call(%w[common common-items drinfomon equipmanager])
+ 
+class Locksmithing
+  include DRC
+  include DRCI
+  def initialize
+    settings = get_settings
+    @equipment_manager = EquipmentManager.new(settings)
+    @daily_trainer ||= settings.have_training_box
```
Comment (rpherbig) [inline locksmithing.lic:13]: 💡 Why the `||=`? This is the first time it's assigned to.

File: locksmithing.lic [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+    @box = settings.picking_lockbox || 'training box'
```
Comment (rpherbig) [inline locksmithing.lic:14]: 💡 Default value shouldn't be necessary since you updated `base.yaml`.

File: profiles/base.yaml [hunk 3]
```diff
@@ -721,10 +721,21 @@ mining_implement: shovel
...
+#https://elanthipedia.play.net/Lich_script_repository#locksmithing
+# use ;pick and pick settings first
+pick_live_boxes: true
+# second option to a daily use lockbox trainer like a training box or harvest bag
+have_training_box: false
+picking_lockbox: training box
+picking_worn_lockbox: false
+# consumable boxes will be pulled from your burgle loot_container and should be kept separate from live boxes.
+# list as many as you want - though it will try from the top down.
+consumable_lockboxes:
```
Comment (rpherbig) [inline profiles/base.yaml:734]: ❌ This should have an entry in `base-empty.yaml`

File: locksmithing.lic (hunk 1 above)
Comment (Hiinky) [reply locksmithing.lic:13]: Ahh some of these were in the base script I started with and I didn't look too close. editing now.

File: profiles/base.yaml (hunk 3 above)
Comment (Hiinky) [reply profiles/base.yaml:734]: added.

File: locksmithing.lic (hunk 2 above)
Comment (Hiinky) [reply locksmithing.lic]: Ahh some of these were in the base script I started with and I didn't look too close. editing now.

**Pull request merged**